lib/types/inet-address.js

```diff
@@ -10,238 +11,260 @@ const utils = require("../utils");
+        parts = value.split(":");
+        if (parts.length < 3) {
+            throw new TypeError(
+                "Value could not be parsed as InetAddress: " + value,
+            );
         }
-        applied = true;
-        for (let j = 0; j < filling; j++) {
-            buffer[offset++] = 0;
-            buffer[offset++] = 0;
+        const buffer = utils.allocBufferUnsafe(16);
+        let filling = 8 - parts.length + 1;
+        let applied = false;
+        let offset = 0;
+        const embeddedIp4 = ipv4Pattern.test(parts[parts.length - 1]);
+        if (embeddedIp4) {
+            // Its IPv6 address with an embedded IPv4 address:
+            // subtract 1 from the potential empty filling as ip4 contains 4 bytes instead of 2 of a ipv6 section
+            filling -= 1;
         }
+        function writeItem(uIntValue) {
+            buffer.writeUInt8(+uIntValue, offset++);
+        }
+        for (let i = 0; i < parts.length; i++) {
+            const item = parts[i];
+            if (item) {
+                if (embeddedIp4 && i === parts.length - 1) {
+                    item.split(".").forEach(writeItem);
+                    break;
+                }
+                buffer.writeUInt16BE(parseInt(item, 16), offset);
+                offset = offset + 2;
+                continue;
+            }
+            //its an empty string
+            if (applied) {
+                //there could be 2 occurrences of empty string
+                filling = 1;
+            }
+            applied = true;
+            for (let j = 0; j < filling; j++) {
+                buffer[offset++] = 0;
+                buffer[offset++] = 0;
+            }
+        }
+        if (embeddedIp4 && !InetAddress.#isValidIPv4Mapped(buffer)) {
+            throw new TypeError(
+                "Only IPv4-Mapped IPv6 addresses are allowed as IPv6 address with embedded IPv4 address",
+            );
+        }
+        return new InetAddress(buffer);
     }
-    if (embeddedIp4 && !isValidIPv4Mapped(buffer)) {
-        throw new TypeError(
-            "Only IPv4-Mapped IPv6 addresses are allowed as IPv6 address with embedded IPv4 address",
+
+    /**
+     * Compares 2 addresses and returns true if the underlying bytes are the same
+     * @param {InetAddress} other
+     * @returns {Boolean}
+     */
+    equals(other) {
+        if (!(other instanceof InetAddress)) {
+            return false;
+        }
+        return (
+            this.buffer.length === other.buffer.length &&
+            this.buffer.toString("hex") === other.buffer.toString("hex")
         );
     }
-    return new InetAddress(buffer);
-};
 
-/**
- * Compares 2 addresses and returns true if the underlying bytes are the same
- * @param {InetAddress} other
- * @returns {Boolean}
- */
-InetAddress.prototype.equals = function (other) {
-    if (!(other instanceof InetAddress)) {
-        return false;
+    /**
+     * Returns the underlying buffer
+     * @returns {Buffer}
```

Comment:
`@return` should be used for methods and `@type` should be used for variables and properties. From the user point of view using setters and getters looks like using properties of class, so that case `@type` should be used but for this case `@return` should be used